libkbfs/init.go

```diff
@@ -639,15 +633,29 @@ func doInit(
 	if err != nil {
 		return nil, fmt.Errorf("problem creating service: %s", err)
 	}
-	if registry := config.MetricsRegistry(); registry != nil {
-		service = NewKeybaseServiceMeasured(service, registry)
+
+	// Initialize Chat client (for file edit notifications).
+	chat, err := keybaseServiceCn.NewChat(config, params, kbCtx, kbfsLog)
+	if err != nil {
+		return nil, fmt.Errorf("problem creating chat: %s", err)
 	}
-	config.SetKeybaseService(service)
+	config.SetChat(chat)
 
-	// Initialize KBPKI client (needed for MD Server).
+	// Initialize KBPKI client (needed for KBFSOps and MD Server).
 	k := NewKBPKIClient(config, kbfsLog)
 	config.SetKBPKI(k)
 
+	kbfsOps := NewKBFSOpsStandard(config)
+	config.SetKBFSOps(kbfsOps)
```

Comment:
If we set this before `keybaseServiceCn.NewChat` above, do we still need those `!= nil` checks in `*ChatRPC`?

Reply:
The changes in this file were to make sure chat is initialized _before_ KBFSOps, because now KBFSOps depends on it.  So unfortunately it's a cyclical dependency and I needed to add the `!= nil` checks.  If you see a better way to structure things (besides building an internal, async notification system), let me know.